packages/components/src/ActionBar/ActionBar.component.js

```diff
@@ -8,90 +8,104 @@ const DISPLAY_MODES = {
+	let wrapperClassNamed = null;
+	if (left) {
+		wrapperClassNamed = classNames(css['navbar-left'], 'navbar-left');
+	} else if (right) {
+		wrapperClassNamed = classNames(css['navbar-right'], 'navbar-right');
 	}
+	return (
+		<div className={wrapperClassNamed}>
+			{ selected > 0 && !right ? (
+				<Count selected={selected} />
+			) : null }
+			{ actions.map((action, index) => {
+				const { displayMode, ...rest } = action;
+				switch (displayMode) {
+				case DISPLAY_MODES.SPLIT_DROPDOWN:
+					return (
+						<ActionSplitDropdown key={index} {...rest} />
+					);
+				case DISPLAY_MODES.BTN_GROUP:
+					return (
+						<Actions key={index} {...rest} />
+					);
+				default:
+					return (
+						<Action key={index} {...rest} />
+					);
+				}
+			}) }
+		</div>
+	);
+}
+SwitchActions.propTypes = {
+	actions: PropTypes.shape(actionsShape).isRequired,
+	left: PropTypes.bool,
+	right: PropTypes.bool,
+	selected: PropTypes.number,
+};
+SwitchActions.defaultProps = {
+	actions: [],
+};
 
-	function renderActionBar() {
-		const { left, right } = getActionsToRender();
-		const actionBar = [];
-		if (left) {
-			actionBar.push(
-				<div key={0} className={classNames(css['navbar-left'], 'navbar-left')}>
-					{ selected > 0 ? renderSelectedCount() : null}
-					{ renderActions(left) }
-				</div>,
-			);
-		}
-		if (right) {
-			actionBar.push(
-				<div key={1} className={classNames(css['navbar-right'], 'navbar-right')}>
-					{ selected > 0 && !left ? renderSelectedCount() : null}
-					{ renderActions(right) }
-				</div>,
-			);
-		}
-		return actionBar;
+function Count({ selected }) {
+	if (!selected) {
+		return null;
 	}
+	return (
+		<span className={classNames(css['tc-actionbar-selected-count'], 'tc-actionbar-selected-count')}>
+			{selected} selected
+		</span>
+	);
+}
+Count.propTypes = {
+	selected: PropTypes.number,
+};
 
+function ActionBar(props) {
+	const { left, right } = getActionsToRender(props);
 	return (
 		<nav className={classNames(css['tc-actionbar-container'], 'tc-actionbar-container', 'nav')}>
-			{ renderActionBar() }
+			{ (left || !!props.selected) && (
+				<SwitchActions key={0} actions={left} selected={props.selected} left />
+			)}
+			{props.children}
+			{ right && (
+				<SwitchActions key={1} actions={right} selected={props.selected} right />
+			)}
 		</nav>
 	);
 }
 
-const actionsShape = {
-	left: PropTypes.arrayOf(PropTypes.oneOfType([
-		PropTypes.shape(Action.propTypes),
-		PropTypes.shape(ActionSplitDropdown.propTypes),
-	])),
-	right: PropTypes.arrayOf(PropTypes.oneOfType([
-		PropTypes.shape(Action.propTypes),
-		PropTypes.shape(ActionSplitDropdown.propTypes),
-	])),
-};
-
 ActionBar.propTypes = {
 	selected: PropTypes.number,
-	actions: PropTypes.shape(actionsShape).isRequired,
-	multiSelectActions: PropTypes.shape(actionsShape),
-};
-
-ActionBar.defaultProps = {
-	actions: [],
+	children: PropTypes.node,
 };
 
 ActionBar.DISPLAY_MODES = DISPLAY_MODES;
+ActionBar.Count = Count;
```

Comment:
This has been exposed to previous TAB.
it's to help you to build your own custom ActionBar if you need in your project by composition.
So all sub components are attached to the main one.

the idea is coming from react-bootstrap which is doing that.